### ISSUE TITLE
When will be new version for wordpress?and how connecting function playbackRates?

### ISSUE THREAD

**Issue by surikat1978** [brightcove-archive/zencoder_video-js-for-wordpress#7]
Good Day! When will be new version for wordpress?
and how connecting function playbackRates (video speed)  correctly in wordpress?

**Comment by robbiegod** [comment]
Should we consider this plugin abandoned?

**Comment by mmcc** [comment]
I've always suggested people use VJS directly instead of using this plugin. I (personally) don't think the abstraction is useful enough to warrant the maintenance liability, and the core team definitely doesn't have the time to keep this up to date with each new release.

Not officially speaking for this project necessarily, maybe someone else on core disagrees, but I would like to consider officially listing this as unmaintained and point people to use VJS directly. The other option is to seek out a new maintainer, but logistically it simply can't be someone from the core team due to time constraints. @heff and @nosecreek: thoughts?

**Comment by mmcc** [comment]
Also, just to note, [there are forks](https://github.com/jlad26/video-js-for-wordpress) that are a bit out of date, but less-so.

Another note on the note, I'm not a Wordpress user, so my opinion on the usefulness of any abstractions is probably of questionable authority.

**Comment by nosecreek** [comment]
I think its fair to consider it abandoned in the sense that I haven't made any updates in a long time and likely won't any time soon. I initially started working on it when I needed it for client's websites. I saw it as a way to give back to the community while doing my job - it may be that there are others in a similar situation who may be willing to maintain it in the future?

I think there is value in having a plugin, but it would be worth thinking about the purpose of the plugin. Should it just be bare-bones functionality? If so, maintenance would be quite easy as it would only require updating a couple lines of code with every update. Should it include more advanced functionality such as vimeo and youtube embeds? That is going to require more work and might be better left to a third-party plugin.

**Comment by robbiegod** [comment]
I'd like it if the plugin did these core things:
1. Loads most recent version of video.js
2. Provides shortcode functionality w/ default values. Allows overrides too.

I don't think we actually need the switches in the wordpress admin to turn features on and off.

**Comment by mmcc** [comment]
Yeah the more I think about it, the more it makes sense for WYSIWYG users to be able to have access to things like this. It's easy to forget the WP end-user is probably not going to be someone that wants to know anything about underlying JS libraries. To be clear, @nosecreek, your work was greatly appreciated! Just trying to deal with the current situation of you (reasonably) not being able to devote time anymore.

It kinda puts me in the middle of the road, though. On one hand a bare-bones use case could just point to the latest minor release on the CDN and not require any updating at all for minor version bumps. However, I think the more advanced functionality is what makes this really useful to your every day user, so that's a really tough call to make. Maybe step 1 is pinging the folks maintaining forks to see if they'd like to take on keeping this up to date?

**Comment by robbiegod** [comment]
Yes, i'd agree with you there.  Pinging the guys that have forks.  They've done some nice work there.  I would definitely first check with them to see if they want to become the maintainers.  It would be nice to see the pull requests actually merged to the master branch :)

And, maybe we can also get the plugin updated on wordpress.org.

If that fails then the next best thing would be a scaled down plugin that is easier to maintain.

**Comment by robbiegod** [comment]
have you heard back from any of the contributors on this issue?

**Comment by stopspazzing** [comment]
Hey, guys. I would suggest slimming down for easier maintenance. The extra feature aren't needed. Personally bare minimum of most commonly used features would be nice, like youtube for instance, but nothing more. I'll just fork it again and add what is needed. :)

**Comment by mmcc** [comment]
@stopspazzing You could fork it again, add what is needed, and then submit a PR back :)

**Comment by stopspazzing** [comment]
@mmcc Sounds good. I'll probably add at minimum youtube.

**Comment by robbiegod** [comment]
Sounds great.  Once this is all completed, would you be able to update the plugin in the wordpress repo?

**Comment by mmcc** [comment]
Yep!

**Comment by robbiegod** [comment]
awesome, thank you!  u guys are awesome!

**Comment by robbiegod** [comment]
@stopspazzing @mmcc How are things going with the update to the plugin?

**Comment by onigetoc** [comment]
I did update it for Videojs 5.4
https://github.com/onigetoc/video-js-for-wordpress

Test page demo: http://wpgit.org/test-videojs
